server/cluster/cluster.go

```diff
@@ -1690,30 +1674,79 @@ func (c *RaftCluster) SetMinResolvedTS(storeID, minResolvedTS uint64) error {
 	return c.putStoreLocked(newStore)
 }
 
-func (c *RaftCluster) runMinResolvedTSJob(saveInterval time.Duration) {
-	defer c.wg.Done()
-	if saveInterval == 0 {
+func (c *RaftCluster) checkAndUpdateMinResolvedTS() {
+	c.Lock()
+	defer c.Unlock()
+
+	if !c.isInitialized() {
+		return
+	}
+	curMinResolvedTS := uint64(math.MaxUint64)
+	for _, s := range c.GetStores() {
+		if !core.IsAvailableForMinResolvedTS(s) {
+			continue
+		}
+		if curMinResolvedTS > s.GetMinResolvedTS() {
+			curMinResolvedTS = s.GetMinResolvedTS()
+		}
+	}
+	if curMinResolvedTS == math.MaxUint64 || curMinResolvedTS <= c.minResolvedTS {
 		return
 	}
+	c.minResolvedTS = curMinResolvedTS
+	c.storage.SaveMinResolvedTS(curMinResolvedTS)
```

Comment:
Can we save storage without holding the lock?

Reply:
Good catch, the `SaveMinResolveTS` is a big operator and here we do needn't the lock, I think to save without holding the lock should be better.